### PR TITLE
Fix/3451 duplicate session upon login

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

File: spec/ParseUser.spec.js [hunk 1]
```diff
@@ -3453,4 +3453,34 @@ describe('Parse.User testing', () => {
       done();
     });
   });
+
+  it('does not duplicate session when loggin in multiple times #3451', (done) => {
```
Comment (montymxb) [inline spec/ParseUser.spec.js]: **loggin** should be *logging

File: spec/ParseUser.spec.js [hunk 2]
```diff
@@ -3453,4 +3453,34 @@ describe('Parse.User testing', () => {
...
+      while(promises.length != 5) {
+        Parse.User.logIn('yolo', 'yolo')
+        promises.push(Parse.User.logIn('yolo', 'yolo').then((res) => {
+          // generate a new session token every times
+          expect(res.getSessionToken()).not.toBe(user.getSessionToken());
+        }));
+      }
+      return Promise.all(promises);
+    }).then(() => {
+      // wait because session destruction is not syncrhonous
```
Comment (montymxb) [inline spec/ParseUser.spec.js]: typo in syncrhonous

File: spec/ParseUser.spec.js [hunk 3]
```diff
@@ -3453,4 +3453,34 @@ describe('Parse.User testing', () => {
...
+    user.signUp({
+      username: 'yolo',
+      password: 'yolo',
+      email: 'yo@lo.com'
+    }).then(() => {
+      const promises = [];
+      while(promises.length != 5) {
+        Parse.User.logIn('yolo', 'yolo')
+        promises.push(Parse.User.logIn('yolo', 'yolo').then((res) => {
+          // generate a new session token every times
```
Comment (montymxb) [inline spec/ParseUser.spec.js]: kinda weird wording in comment but not that big of a deal

**Comment by flovilmart** [comment]
we / I need to refactor at some point UserRouter (too many things) and Session creation as spread all across the project (lot of duplicated boilerplate).

**Comment by montymxb** [comment]
Yeah, some consolidation is long overdue! I would be more than willing to help out with that, good amount that could be cleaned up. Lot to expect from one person ☝️.

**Pull request merged**